packages/nextra-theme-docs/src/components/toc.tsx

```diff
@@ -14,7 +14,7 @@ export type TOCProps = {
 }
 
 export function TOC({ headings, filePath }: TOCProps): ReactElement {
-  const slugger = new Slugger()
+  const slugger = useMemo(() => new Slugger(), [])
```

Comment:
I guess even better will be to instantiate class outside of the function declaration

Reply:
What’s the difference with useMemo?

Reply:
useMemo cant provide a stale reference here.

```
You may rely on useMemo as a performance optimization, not as a semantic guarantee. 
In the future, React may choose to “forget” some previously memoized values and recalculate them on next render, 
e.g. to free memory for offscreen components. 
Write your code so that it still works without useMemo — and then add it to optimize performance.
```

Reply:
@promer94 you are right, I'll fix it, wrote a selector to ban `useMemo(…, [])` usage

<img width="1004" alt="image" src="https://user-images.githubusercontent.com/7361780/211334147-5d4b5d97-bd2f-496c-bf57-181ec75dc465.png">